src/axom/core/tests/core_array.hpp

```diff
@@ -1906,4 +1906,26 @@ TEST(core_array, checkConstConversion)
   // v_double_2d_view_copy(0,0) = 1.1; // Fails to compile as it should
 }
 
+//------------------------------------------------------------------------------
+
+TEST(core_array, checkVariadicCtors)
+{
+  int i = 5;
+  size_t s = 6;
+
+  Array<int, 2> arr1(i, i);
```

Comment:
Gut check -- will these cause "-Wunused` warnings (or their equivalent?

If so, we might need to add `AXOM_UNUSED_VAR(arr{i})` for each of them.

https://github.com/LLNL/axom/blob/0ae8969805be334bd310d1ed88c429e44034ba10/src/axom/core/Macros.hpp#L152-L172

Reply:
These don't trigger warnings about unused variables on my local build.

`i` and `s` are used in the ctors, and I don't think the compiler can warn about complicated objects like `axom::Array` being unused, because it's not a trivial type, see: 

https://godbolt.org/z/87a3Y93aM

------------

also, I ran `make style`, but it still seems to be failing the style check, so I might need someone else to apply the formatting, since my tools don't seem to be compatible.

Reply:
@samuelpmishLLNL I think you may need to run make style with clang 10. @white238 is that correct?

Reply:
I merged develop and ran style.